src/components/ModalWithForm.js

```diff
@@ -0,0 +1,78 @@
+import Modal from "./Modal.js";
+
+export default class ModalWithForm extends Modal {
+  constructor({ modalSelector, onSubmitCallback }) {
+    super({ modalSelector });
+
+    //note: onSubmitCallback must always return a promise.
+    this._onSubmitCallback = onSubmitCallback;
+    this._submitButton = this._modalElement.querySelector(
+      ".modal__submit-button"
+    );
+    this._inputFields = this._modalElement.querySelectorAll(".modal__input");
+    this._form = this._modalElement.querySelector(".modal__form");
+    this._currentProfileName = document.querySelector(".profile__name");
```

Comment:
```js

   this._currentProfileName = document.querySelector(".profile__name");
    this._currentProfileDescription = document.querySelector(
      ".profile__description"
    );
```

Please, delete them because you don't use them

And `ModalWithForm` is a universal class for any modals with a form - it should not contain hardcoded data